### PR TITLE
add a portal style login and admin configuration

### PR DESCRIPTION
**Pull request by briancwald**
screenshot 2017-07-13 15 09 12

### CODE REVIEW

File: config/install/block.block.material_admin_sitebranding.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  label: 'Site branding'
+  provider: system
+  label_display: '0'
+  use_site_logo: true
+  use_site_name: false
+  use_site_slogan: false
+visibility:
+  request_path:
+    id: request_path
+    pages: "/user/login\r\n/user/password\r\n/user/register"
```
Comment (mortenson) [inline config/install/block.block.material_admin_sitebranding.yml:23]: Should we add /user/reset here as well?

**Comment by mortenson** [comment]
I like this - although /user pages usually use the frontend theme. It's configurable which is good, but if you weren't using a material-based frontend theme and an end-user went to log in and saw this style of page, it might be a bit awkward.

**Comment by e0ipso** [comment]
I can't wait so see this merged.

**Comment by briancwald** [comment]
@mortenson yeah I mostly built this for the decoupled audience that only use Drupal as an authoring tool, so this way there is no need to have a frontend theme.

**Comment by briancwald** [comment]
It doesnt show on /user/1 or any of those, its explicit to just login/register/password

File: material_admin.theme [hunk 2]
```diff
@@ -162,3 +167,14 @@ function material_admin_preprocess_links__dropbutton(&$variables) {
 function material_admin_preprocess_color_scheme_form(&$variables) {
   $variables['#attached']['library'][] =  'material_admin/color';
 }
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function material_admin_theme_suggestions_page_alter(array &$suggestions, array $variables) {
+  $route_match = \Drupal::routeMatch()->getRouteName();
+  $portal_paths = array('user.pass', 'user.login', 'user.register');
```
Comment (briancwald) [inline material_admin.theme:176]: @mortenson this is the method I came up with here, it's pretty manual.. but I couldn't think of a better way without it showing this template for all /user/* which I don't want to do.

Comment (mortenson) [reply material_admin.theme:176]: This seems OK - not sure how caching plays into it but I can't imagine this operation is too costly.

Comment (mortenson) [reply material_admin.theme:176]: You should check for the theme setting here as well, right? From what I can see it's only used to determine if a class is applied.

Comment (briancwald) [reply material_admin.theme:176]: Yes I must have removed that when I was testing an issue and forgot to put it back, thanks!

**Comment by briancwald** [comment]
I think this is good to go now, fresh install:

- Confirmed it shows and works for just the login, register, reset page.
- Confirmed the configuration settings work.
- Confirmed positioning and config install.


@mortenson did you try it out by chance?

**Comment by e0ipso** [comment]
Can you push this code to the drupal.org repository?

**Comment by briancwald** [comment]
I checked and it's all there, does it not work for you? maybe there is a different issue.

**Pull request merged**